data-raw/jhu_csse_daily_subset.R

```diff
@@ -2,60 +2,88 @@ library(epidatr)
+  complete(geo_value, time_value = full_seq(time_value, 1L))
 
-deaths_7dav_incidence_prop <- pub_covidcast(
+deaths_incidence_num <- pub_covidcast(
   source = "jhu-csse",
-  signals = "deaths_7dav_incidence_prop",
+  signals = "deaths_incidence_num",
   geo_type = "state",
   time_type = "day",
   geo_values = "ca,fl,ny,tx,ga,pa",
-  time_values = epirange(20200301, 20211231),
+  time_values = epirange(20200301, 20211231)
 ) %>%
-  select(geo_value, time_value, death_rate_7d_av = value) %>%
-  arrange(geo_value, time_value)
+  select(geo_value, time_value, deaths = value) %>%
+  complete(geo_value, time_value = full_seq(time_value, 1L))
 
-confirmed_incidence_num <- pub_covidcast(
+confirmed_incidence_prop <- pub_covidcast(
   source = "jhu-csse",
-  signals = "confirmed_incidence_num",
+  signals = "confirmed_incidence_prop",
   geo_type = "state",
   time_type = "day",
   geo_values = "ca,fl,ny,tx,ga,pa",
-  time_values = epirange(20200301, 20211231),
+  time_values = epirange(20200301, 20211231)
 ) %>%
-  select(geo_value, time_value, cases = value) %>%
-  arrange(geo_value, time_value)
+  select(geo_value, time_value, case_rate = value) %>%
+  complete(geo_value, time_value = full_seq(time_value, 1L))
 
-confirmed_7dav_incidence_num <- pub_covidcast(
+deaths_incidence_prop <- pub_covidcast(
   source = "jhu-csse",
-  signals = "confirmed_7dav_incidence_num",
+  signals = "deaths_incidence_prop",
   geo_type = "state",
   time_type = "day",
   geo_values = "ca,fl,ny,tx,ga,pa",
-  time_values = epirange(20200301, 20211231),
+  time_values = epirange(20200301, 20211231)
 ) %>%
-  select(geo_value, time_value, cases_7d_av = value) %>%
-  arrange(geo_value, time_value)
+  select(geo_value, time_value, death_rate = value) %>%
+  complete(geo_value, time_value = full_seq(time_value, 1L))
+
 
-jhu_csse_daily_subset <- confirmed_7dav_incidence_prop %>%
-  full_join(deaths_7dav_incidence_prop,
+joined <- confirmed_incidence_num %>%
+  full_join(deaths_incidence_num,
     by = c("geo_value", "time_value")
   ) %>%
-  full_join(confirmed_incidence_num,
+  full_join(confirmed_incidence_prop,
     by = c("geo_value", "time_value")
   ) %>%
-  full_join(confirmed_7dav_incidence_num,
+  full_join(deaths_incidence_prop,
     by = c("geo_value", "time_value")
   ) %>%
-  as_epi_df()
+  as_epi_df() %>%
+  arrange(geo_value, time_value)
+
+jhu_csse_daily_subset <- joined %>%
+  group_by(geo_value) %>%
+  epi_slide(
+    cases_7d_av = sum(cases) / 7,
+    before = 7L
+  ) %>%
+  ungroup() %>%
+  group_by(geo_value) %>%
+  epi_slide(
+    case_rate_7d_av = sum(case_rate) / 7,
+    before = 7L
+  ) %>%
+  ungroup() %>%
+  group_by(geo_value) %>%
+  epi_slide(
+    deaths_7d_av = sum(deaths) / 7,
+    before = 7L
+  ) %>%
+  ungroup() %>%
+  group_by(geo_value) %>%
+  epi_slide(
+    death_rate_7d_av = sum(death_rate) / 7,
+    before = 7L
+  ) %>%
+  ungroup()
 
 usethis::use_data(jhu_csse_daily_subset, overwrite = TRUE)
```

Comment:
I'm getting weird results here.  It seems like the current rda is simply from regenerating with the old code (which I [thought](https://github.com/cmu-delphi/epiprocess/issues/251#issuecomment-1334642732) would resolve this).  But running re-running the script on this branch gives me different results than the rda.

1. Different set of column names; this might break vignettes if we try to regenerate the data in the future.
```
waldo::compare(branch, branch_regen)waldo::compare(branch, branch_regen)


`old` is length 6
`new` is length 10

    names(old)         | names(new)             
[1] "geo_value"        | "geo_value"        [1] 
[2] "time_value"       | "time_value"       [2] 
[3] "case_rate_7d_av"  - "cases"            [3] 
[4] "death_rate_7d_av" - "deaths"           [4] 
[5] "cases"            - "case_rate"        [5] 
[6] "cases_7d_av"      - "death_rate"       [6] 
                       - "cases_7d_av"      [7] 
                       - "case_rate_7d_av"  [8] 
                       - "deaths_7d_av"     [9] 
                       - "death_rate_7d_av" [10]

`attr(old, 'metadata')$as_of`: "2024-01-25 11:32:25"
`attr(new, 'metadata')$as_of`: "2024-01-25 11:35:02"

     old$case_rate_7d_av   | new$case_rate_7d_av                    
[.... numerical differences, not sure if waldo does this right or not so trying a join approach below]
```

2. Checking for numerical differences (due to either API issues or `epi_slide` pitfalls):
```
full_join(branch, branch_regen, by = c("geo_value", "time_value")) %>%
  mutate(case7dav_discrep = abs(case_rate_7d_av.x - case_rate_7d_av.y)*200/(abs(case_rate_7d_av.x) + abs(case_rate_7d_av.y) + 1e-1)) %>%
  arrange(desc(case7dav_discrep)) %>%
  select(geo_value, time_value, cases.x, cases_7d_av.x, cases_7d_av.y)
An `epi_df` object, 4,026 x 5 with metadata:
* geo_type  = state
* time_type = day
* as_of     = 2024-01-25 11:32:25

# A tibble: 4,026 × 5
   geo_value time_value cases.x cases_7d_av.x cases_7d_av.y
 * <chr>     <date>       <dbl>         <dbl>         <dbl>
 1 fl        2021-11-26       0            0          1413 
 2 ga        2020-11-10    4319         2447.         6962.
 3 fl        2021-10-08   25184         3598.         8926.
 4 fl        2021-10-01   37299         5328.        12663 
 5 fl        2021-10-15   18807         2687.         6284.
 6 fl        2021-10-22   14564         2081.         4767.
 7 fl        2021-10-29   12151         1736.         3816.
 8 fl        2021-11-05   10498         1500.         3236.
 9 fl        2021-06-18   10095         1442.         3078.
10 fl        2021-11-19    9891         1413          2885.
# ℹ 4,016 more rows
# ℹ Use `print(n = ...)` to see more rows

full_join(branch, branch_regen, by = c("geo_value", "time_value")) %>%
  mutate(case7dav_discrep = abs(case_rate_7d_av.x - case_rate_7d_av.y)*200/(abs(case_rate_7d_av.x) + abs(case_rate_7d_av.y) + 1e-1)) %>%
  filter(geo_value == "fl", time_value %>% between(as.Date("2021-11-26")-6L, as.Date("2021-11-26"))) %>%
  select(geo_value, time_value, cases.x, cases_7d_av.x, cases_7d_av.y)
+ An `epi_df` object, 7 x 5 with metadata:
* geo_type  = state
* time_type = day
* as_of     = 2024-01-25 11:32:25

# A tibble: 7 × 5
  geo_value time_value cases.x cases_7d_av.x cases_7d_av.y
* <chr>     <date>       <dbl>         <dbl>         <dbl>
1 fl        2021-11-20       0          1413          1413
2 fl        2021-11-21       0          1413          1413
3 fl        2021-11-22       0          1413          1413
4 fl        2021-11-23       0          1413          1413
5 fl        2021-11-24       0          1413          1413
6 fl        2021-11-25       0          1413          1413
7 fl        2021-11-26       0             0          1413
```
Oh, this one is just from `before = 7L` instead of `before = 6L`.  But I expect there might be some other errors on the first 6 time_values, since we didn't download the necessary data to calculate the averages here, nor NA-d/filtered them out.
```
full_join(branch, branch_regen, by = c("geo_value", "time_value")) %>%
  mutate(case7dav_discrep = abs(case_rate_7d_av.x - case_rate_7d_av.y)*200/(abs(case_rate_7d_av.x) + abs(case_rate_7d_av.y) + 1e-1)) %>%
  filter(geo_value == "ca", time_value %>% between(as.Date("2020-03-01"), as.Date("2020-03-07"))) %>%
  select(geo_value, time_value, cases.x, cases_7d_av.x, cases_7d_av.y)
+ An `epi_df` object, 7 x 5 with metadata:
* geo_type  = state
* time_type = day
* as_of     = 2024-01-25 11:32:25

# A tibble: 7 × 5
  geo_value time_value cases.x cases_7d_av.x cases_7d_av.y
* <chr>     <date>       <dbl>         <dbl>         <dbl>
1 ca        2020-03-01       6          1.29         0.857
2 ca        2020-03-02       4          1.71         1.43 
3 ca        2020-03-03       6          2.43         2.29 
4 ca        2020-03-04      11          3.86         3.86 
5 ca        2020-03-05      10          5.29         5.29 
6 ca        2020-03-06      18          7.86         7.86 
7 ca        2020-03-07      26         11.6         11.6  
```